From: Sergio Costas Rodriguez <sergio.costas@canonical.com>
Date: Fri, 2 Feb 2024 17:30:16 +0100
Subject: snap-policy: fix memory leak

This patch fixes a memory leak when there is a snap with an
invalid ID.

Forwarded: https://gitlab.freedesktop.org/pipewire/pipewire/-/merge_requests/1883
---
 src/modules/module-protocol-pulse/snap-policy.c | 1 +
 1 file changed, 1 insertion(+)

--- a/src/modules/module-protocol-pulse/snap-policy.c
+++ b/src/modules/module-protocol-pulse/snap-policy.c
@@ -83,6 +83,7 @@
 	snap_id = g_strdup(aa_label + strlen(SNAP_LABEL_PREFIX));
 	separator = strchr(snap_id, '.');
 	if (separator == NULL) {
+		g_free(snap_id);
 		pw_log_info("snap_get_audio_permissions: aa_label has only one dot; not a valid ID.");
 		return PW_SANDBOX_ACCESS_NONE;
 	}
